### DISCUSSION TITLE
Typescript and/or type annotations

### DISCUSSION THREAD

**Discussion by xkortex** [MaxRobinsonTheGreat/EvolutionSimulatorV2#51] [Ideas and Suggestions]
How do you feel about transitioning the project to Typescript? It can be done progressively over time; you can include `.js` files in your `.ts` so that the conversion doesn't need to happen at once. 

It helps catch a wide variety of errors and makes the codebase more maintainable. All the cool kids are doing it. It looks good on resumes ;) 

You can also [annotate docstrings](https://ricostacruz.com/til/typescript-jsdoc) which gives you some type hinting and improves interop with TS.

**Comment by MaxRobinsonTheGreat** [comment]
It would certainly be nice, but would require time to learn and implement, which I don't have much of right now. 

If I were to dedicate time to architectural improvements, I would use it to replace jquery/controller classes with vue components, which I think would pay itself back more than typescript.

**Reply by xkortex** [reply]
That's legit. :)

**Reply by xkortex** [reply]
Do you think a Vue conversion could be done progressively, replacing various parts of the jquery UI with Vue components, or is it kind of a rip-the-bandaid-off affair?

**Reply by MaxRobinsonTheGreat** [reply]
It can be done progressively, but its a small enough project that I could probably do it all at once if I dedicated a couple weekends to it. We'll see

**Reply by xkortex** [reply]
Cool. LMK if you'd like some help with it. I'm trying to get better at JS and learn some Vue and this seems like a fun project to cut my teeth on.
